pkg/stub/handler.go

```diff
@@ -81,6 +81,14 @@ type Handler struct {
 	samplesResource *v1alpha1.SamplesResource
 	registrySecret  *corev1.Secret
 
+	// we maintain a separate SamplesResource cache
+	// entry when this is a default rhel install and we
+	// are waiting for the credential since we are
+	// in an erorr state; our current approach is
+	// to only update the standard samplesResource
+	// cache entry for valid / non-error states
+	waitingForCredential *v1alpha1.SamplesResource
```

Comment:
is there a reason this is better than just setting a "waitingForCredential" boolean value, and holding the pending samplesResource in the existing "samplesResource" field?

Or setting a waitingForCredential boolean and when the credential shows up, simply fetching the current SamplesResource and processing it (after having processed the credential of course)?

Reply:
the complete separation of concerns was cleaner and easier to keep separate in my mind this way 

it wanted to keep samplesResource as only ever for "good" samplesresource